Cognite.Simulator.Utils/ModelLibraryBase.cs

```diff
@@ -234,23 +231,68 @@ private async Task<T> TryReadRemoteModelRevision(string modelRevisionExternalId,
             return null;
         }
 
-        /// <inheritdoc/>
-        public async Task<T> GetModelRevision(string modelRevisionExternalId)
+        /// <summary>
+        /// If model revision is not found in the local state, this method will try to read it from CDF, download the file and extract the model information.
+        /// If the model revision is already in the local state and has not been changed, it will return the existing state.
```

Comment:
This doc comment isn't true, I think? You unconditionally call `TryReadRemoteModelRevision` on the first line, which will make a call to CDF.

Wouldn't it be better to cache by external ID, and look that up first?

Reply:
fixed the wording